### ISSUE TITLE
Canny edge penalty mask for loss calculation

### ISSUE THREAD

**Issue by alexbarnett12** [appliedinnovation/fast-depth#23]
This [paper](https://arxiv.org/pdf/2007.11256.pdf) presents results of training a MDE network with an additional canny edge penalty mask on the loss function. This mask adds a constant weight to all pixels within a certain range of the canny edges in the input. This is a great idea for teaching the net to focus on sharp lines and discontinuities, and their examples show results with marked improvement in depth predictions at edges. Luckily for me, they _have not uploaded_ their canny edge detection code. To make training take anything less then 4000 years, I am going to have to write a PyTorch GPU implementation of the canny edge detection algorithm (woohoo). This may or may not be awful - this [repo](https://github.com/DCurro/CannyEdgePytorch) will be a great starting point, but they have not implemented the whole algorithm, such as the hysteresis part.
